pkg/sources/gitlab/gitlab.go

```diff
@@ -406,8 +428,17 @@ func (s *Source) getAllProjects(ctx context.Context, apiClient *gitlab.Client) (
 	return projects, nil
 }
 
-func (s *Source) getReposFromGitlab(ctx context.Context, apiClient *gitlab.Client, ignoreRepo func(repo string) bool) ([]string, error) {
-	projects, err := s.getAllProjects(ctx, apiClient)
+// getReposFromGitlab enumerates all repositories of all projects in GitLab
+// using the provided API client. Ignored repositories will be skipped. The
+// reporter, if provided, is used for reporting errors that would only
+// otherwise be logged, and reporting repositories as they are found.
+func (s *Source) getReposFromGitlab(
+	ctx context.Context,
+	apiClient *gitlab.Client,
+	ignoreRepo func(string) bool,
+	reporter sources.UnitReporter,
+) ([]string, error) {
+	projects, err := s.getAllProjects(ctx, apiClient, reporter)
```

Comment:
It would be nice if this could stream the projects so we can start getting the repos early, but I'll leave that for a future optimization.